app/org/sagebionetworks/bridge/dynamodb/DynamoTaskEvent.java

```diff
@@ -77,18 +86,19 @@ public Builder withEventType(TaskEventType type) {
             this.eventType = type;
             return this;
         }
-        public Builder withValue(String value) {
-            this.value = value;
+        public Builder withAnswerValue(String answerValue) {
+            this.answerValue = answerValue;
             return this;
         }
         private String getEventId() {
             if (type == null) {
                 return null;
             }
+            if (type == TaskEventObjectType.ENROLLMENT) {
+                return type.name().toLowerCase();
+            }
             String typeName = type.name().toLowerCase();
-            if (objectId != null && eventType != null && value != null) {
```

Comment:
Could we ever have a scenario where objectId is null while eventType or value is null?

I was expecting something like this without the if-else's

```
    final String objectIdVal = objectId == null ? "" : objectId;
    final String eventTypeVal = value == null ? "" : eventType.name().toLowerCase() + "=" + value;
    ...
    typeName = String.format("%s:%s:%s", typeName, objectIdVal, eventTypeVal);
```